app-modules/engagement/src/Actions/DeliverEngagements.php

```diff
@@ -68,6 +68,6 @@ public function handle(): void
 
     public function middleware(): array
     {
-        return [new WithoutOverlapping(Tenant::current()->id)];
+        return [(new WithoutOverlapping(Tenant::current()->id))->releaseAfter(30)];
```

Comment:
I don't think this is what we want to do here. This job is dispatched onto the queue by the scheduler every minute and we don't really need a job that encounters a lock to be retried because of this.

So I propose we remove this `releaseAfter` and add the following:
- `expireAfter` with a timeout of something like 3 minutes to prevent issues if the previous job did not clear the lock.
- `dontRelease` so that a job that encounters a lock does not go back onto the queue.

What do you think?

Reply:
Makes sense - this has been adjusted.